bundles/org.openhab.binding.hdpowerview/src/main/java/org/openhab/binding/hdpowerview/internal/handler/HDPowerViewShadeHandler.java

```diff
@@ -12,119 +12,253 @@
  */
 package org.openhab.binding.hdpowerview.internal.handler;
 
-import java.io.IOException;
+import static org.openhab.binding.hdpowerview.internal.HDPowerViewBindingConstants.*;
+import static org.openhab.binding.hdpowerview.internal.api.ActuatorClass.*;
+import static org.openhab.binding.hdpowerview.internal.api.CoordinateSystem.*;
 
-import javax.ws.rs.core.Response;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
 
+import javax.ws.rs.ProcessingException;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
 import org.eclipse.smarthome.core.library.types.OnOffType;
 import org.eclipse.smarthome.core.library.types.PercentType;
 import org.eclipse.smarthome.core.library.types.StopMoveType;
 import org.eclipse.smarthome.core.library.types.UpDownType;
 import org.eclipse.smarthome.core.thing.ChannelUID;
 import org.eclipse.smarthome.core.thing.Thing;
 import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
 import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.core.types.RefreshType;
+import org.eclipse.smarthome.core.types.State;
 import org.eclipse.smarthome.core.types.UnDefType;
-import org.openhab.binding.hdpowerview.internal.HDPowerViewBindingConstants;
+import org.openhab.binding.hdpowerview.internal.HDPowerViewWebTargets;
+import org.openhab.binding.hdpowerview.internal.HubMaintenanceException;
+import org.openhab.binding.hdpowerview.internal.api.ActuatorClass;
+import org.openhab.binding.hdpowerview.internal.api.CoordinateSystem;
 import org.openhab.binding.hdpowerview.internal.api.ShadePosition;
-import org.openhab.binding.hdpowerview.internal.api.responses.Shades.Shade;
+import org.openhab.binding.hdpowerview.internal.api.responses.Shade;
+import org.openhab.binding.hdpowerview.internal.api.responses.Shades.ShadeData;
 import org.openhab.binding.hdpowerview.internal.config.HDPowerViewShadeConfiguration;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 /**
- * Handles commands for an HD Power View shade
+ * Handles commands for an HD PowerView Shade
  *
  * @author Andy Lintner - Initial contribution
+ * @author Andrew Fiddian-Green - Added support for secondary rail positions
  */
+@NonNullByDefault
 public class HDPowerViewShadeHandler extends AbstractHubbedThingHandler {
 
-    private static final int MAX_POSITION = 65535;
-    private static final int MAX_VANE = 32767;
-
     private final Logger logger = LoggerFactory.getLogger(HDPowerViewShadeHandler.class);
 
+    private static final int REFRESH_DELAY = 10;
```

Comment:
It's good practice to append the unit to the field name e.g. REFRESH_DELAY_SEC.

Reply:
Ok. Will do.